core/formats/formats_burst_trie.cpp

```diff
@@ -3121,8 +3112,54 @@ class field_reader final : public irs::field_reader {
         owner_->terms_in_cipher_.get(), *fst_);
     }
 
-    virtual size_t bit_union(const cookie_provider& provider,
-                             size_t* set) const override {
+    term_meta term(bytes_view term) const override {
+      single_term_iterator<FST> it{meta(), *owner_->pr_,
+                                   owner_->terms_in_->reopen(),
+                                   owner_->terms_in_cipher_.get(), *fst_};
+
+      it.seek(term);
+      return it.meta();
+    }
+
+    size_t read_documents(bytes_view term, doc_id_t* docs,
+                          size_t count) const override {
+      if (IRS_UNLIKELY(!count || !docs)) {
+        return 0;
+      }
+
+      single_term_iterator<FST> it{meta(), *owner_->pr_,
+                                   owner_->terms_in_->reopen(),
+                                   owner_->terms_in_cipher_.get(), *fst_};
+      if (!it.seek(term)) {
+        return 0;
+      }
+
+      if (const auto& meta = it.meta(); meta.docs_count == 1) {
+        *docs = doc_limits::min() + meta.e_single_doc;
+        return 1;
+      }
+
+      auto docs_it = it.postings(IndexFeatures::NONE);
+
+      const auto* doc = irs::get<document>(*docs_it);
+
+      if (IRS_UNLIKELY(!doc)) {
+        assert(false);
+        return 0;
+      }
+
+      auto* p = docs;
+
+      while (count && docs_it->next()) {
+        *p++ = doc->value;
+        --count;
+      }
+
+      return std::distance(docs, p);
```

Comment:
NIT
instead of this you can use for `(; iter < count && docs_it->next(); ++iter)`